apps/backend/src/deposits/cash-deposit.service.ts

```diff
@@ -169,4 +170,68 @@ export class CashDepositService {
       },
     });
   }
+  /**
+   * Find all cash deposits for the details report - return any entities marked as matched or in progress on this day as well as any pending
+   * @param dateRange
+   * @param program
+   * @returns
+   */
+  async findCashDepositsForDetailsReport(
+    dateRange: DateRange,
+    program: Ministries
+  ): Promise<CashDepositEntity[]> {
+    const reconciled = await this.cashDepositRepo.find({
+      where: {
+        reconciled_on: Between(
+          parse(dateRange.minDate, 'yyyy-MM-dd', new Date()),
+          parse(dateRange.maxDate, 'yyyy-MM-dd', new Date())
+        ),
+        metadata: { program },
+        status: In([MatchStatus.EXCEPTION, MatchStatus.MATCH]),
+      },
+      order: {
+        pt_location_id: 'ASC',
+        reconciled_on: 'ASC',
+        deposit_amt_cdn: 'ASC',
+        status: 'ASC',
+      },
+    });
+    const in_progress = await this.cashDepositRepo.find({
+      where: {
+        in_progress_on: Between(
+          parse(dateRange.minDate, 'yyyy-MM-dd', new Date()),
+          parse(dateRange.maxDate, 'yyyy-MM-dd', new Date())
+        ),
+        metadata: { program },
+        status: MatchStatus.IN_PROGRESS,
+      },
+      order: {
+        pt_location_id: 'ASC',
+        in_progress_on: 'ASC',
+        deposit_amt_cdn: 'ASC',
+      },
+      relations: {
+        payment_matches: false,
+      },
+    });
+    const { minDate, maxDate } = dateRange;
+    const pending = await this.cashDepositRepo.find({
+      where: {
+        deposit_date: Raw(
```

Comment:
'Between' was having an issue here. The deposit_date is actually a string thanks to the typeorm weirdness with the 'date' type columns. (Stored in db as a date, but when type: 'date' is specified on the column decorator typeorm treats it as a string... I think this is because the date type column is only a date, not a date + time, so not a timestamp)